### PR TITLE
Improved error handling on submit

### PR DESCRIPTION
**Pull request by srh-sloan**

### CODE REVIEW

File: app/default/routes.py [hunk 1]
```diff
@@ -287,14 +288,15 @@ def submit_application():
...
-    current_app.logger.warning(f"Encountered 404: {error}")
+    current_app.logger.warning(f"Encountered 404 against url {request.path}: {error}")
     round_data = get_round_data_fail_gracefully(
         Config.DEFAULT_FUND_ID, Config.DEFAULT_ROUND_ID
     )
     return render_template("404.html", round_data=round_data), 404
 
 
 @default_bp.errorhandler(500)
+@default_bp.errorhandler(Exception)
```
Comment (gidsg) [inline app/default/routes.py:299]: What effect does this extra decorator have?

Comment (srh-sloan) [reply app/default/routes.py:299]: It means if an exception is thrown rather than explicitly using `abort(500)`, this error handler still gets called

Comment (srh-sloan) [reply app/default/routes.py:299]: So on line 268, we get the branded error page rather than a generic one (or the debug view in development)

**Review by gio-karunakaran — APPROVED** [review]
LGTM 👍🏾

**Review by gidsg — APPROVED** [review]

**Pull request merged**